src/com/zebrunner/jenkins/pipeline/runner/AbstractRunner.groovy

```diff
@@ -16,8 +19,24 @@ public abstract class AbstractRunner extends BaseObject {
 
     public AbstractRunner(context) {
         super(context)
-        
         sc = new SonarClient(context)
+
+        def host = Configuration.get("GITHUB_HOST")
+        def org = Configuration.get("GITHUB_ORGANIZATION")
+        def repo = Configuration.get("repo")
+        def branch = Configuration.get("branch")
+
+        switch (host.toLowerCase()) {
```

Comment:
copy-paste of the switch above? any way to combine this condition in single place?

Reply:
This one was a tricky one at least for me, the old approach was to have in the BaseObject a scm property(which was initialised in the BaseObject constructor and then propagate to all of the classes that extends from the BaseObject. The problem was that now we need to support multiples scm systems per organisation. From the AbstractRunner(or any of his children's) perspective if we don't set the corresponding scm class with the matching repository  it will default to github(the one initialised in BaseObject ). And as there is no connection(OOP connection at least) we need to instantiate the corresponding scm class. i found a way of do in it by using the build args(RegisterRepository) so the TestNG, gradle Runner, docker Runner, etc will have the matching scm with values from the registerRepository job(host, org, repo, branch) and with that the webhook config is covered(no need to modify anything from the runners point of view)